### PR TITLE
xDS: update GCP auth filter config for compatibility with gRPC

### PR DESCRIPTION
**Pull request by markdroth**
Commit Message: xDS: update GCP auth filter config for compatibility with gRPC
Additional Description: gRPC is implementing support for this filter.  However, most gRPC implementations will delegate to the existing google auth libraries for the actual token functionality, and those libraries do not allow configuring many of the parameters present in this configuration.  This PR updates the config accordingly.
Risk Level: Low
Testing: N/A
Docs Changes: Included in PR
Release Notes: N/A
Platform Specific Features: N/A

CC @tyxia

### CODE REVIEW

File: api/envoy/extensions/filters/http/gcp_authn/v3/gcp_authn.proto
Comment (tyxia) [inline api/envoy/extensions/filters/http/gcp_authn/v3/gcp_authn.proto:49]: Looks like there could be OSS users using this field like here https://github.com/envoyproxy/envoy/issues/28894#issuecomment-1675159504. If gRPC doesn't need this field, can gRPC just leave this unset?

I will defer this to API reviewer @htuch

**Review by tyxia — COMMENTED** [review]
LGTM, modulo one comment (deferred to api-shepherd)
Thanks!

File: api/envoy/extensions/filters/http/gcp_authn/v3/gcp_authn.proto
Comment (htuch) [reply api/envoy/extensions/filters/http/gcp_authn/v3/gcp_authn.proto:49]: Yeah, not sure why we are deprecating fields based purely on what a given library supports.

Comment (markdroth) [reply api/envoy/extensions/filters/http/gcp_authn/v3/gcp_authn.proto]: This is less about what individual data planes may implement and more about making this API consistent with that of the google auth libraries.  Most gRPC implementations will wind up delegating to those existing auth libraries rather than reimplementing this functionality themselves, and I would expect most other data planes to do the same thing.  (If it weren't for the fact that C++ is one of the few languages for which there is no google auth library, I would have expected Envoy to have done the same thing, in which case this issue would never have come up.)

That having been said, I feel less strongly about this for `token_header` than I do for `http_uri`.  I'll go ahead and un-deprecate this field but add a comment that not all data planes support it, and we will not support it in gRPC.

File: api/envoy/extensions/filters/http/gcp_authn/v3/gcp_authn.proto
Comment (tyxia) [inline api/envoy/extensions/filters/http/gcp_authn/v3/gcp_authn.proto:34]: One CI failure I noticed is that you will need to have double backticks like `cluster` --> ``cluster``

**Review by tyxia — APPROVED** [review]
LGTM, please fix the CI failure.

Thanks!

File: api/envoy/extensions/filters/http/gcp_authn/v3/gcp_authn.proto
Comment (markdroth) [reply api/envoy/extensions/filters/http/gcp_authn/v3/gcp_authn.proto]: Thanks, fixed.

**Review by htuch — COMMENTED** [review]
/lgtm api

Defer to @tyxia @yanavlasov for review. If we are marking this field deprecated @markdroth, we need the Envoy-side changes to be made IMHO, otherwise we have deprecated and not-ready-yet situation, and people will (and should) be using the non-deprecated option by default.

**Comment by markdroth** [comment]
> If we are marking this field deprecated @markdroth, we need the Envoy-side changes to be made IMHO, otherwise we have deprecated and not-ready-yet situation, and people will (and should) be using the non-deprecated option by default.

Agreed.  @tyxia, can you prepare the necessary code changes?  We can put the API change and the code change together into a single PR.

**Comment by tyxia** [comment]
> > If we are marking this field deprecated @markdroth, we need the Envoy-side changes to be made IMHO, otherwise we have deprecated and not-ready-yet situation, and people will (and should) be using the non-deprecated option by default.
> 
> Agreed. @tyxia, can you prepare the necessary code changes? We can put the API change and the code change together into a single PR.

Sure I will do that.

**Comment by KBaichoo** [comment]
Closing this as I think it is superseded by https://github.com/envoyproxy/envoy/pull/35173. If not the case, lmk and I'll re-open.

**Pull request closed without merging**
